vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -164,6 +174,34 @@ void AdsImpl::InitializeStep4(
   ads_serve_->DownloadCatalog();
 }
 
+#if defined(OS_ANDROID)
+void AdsImpl::RemoveAllNotificationsAfterReboot() {
+  //ads notifications don't sustain reboot, so remove all
+  auto ads_shown_history = client_->GetAdsShownHistory();
+  if (!ads_shown_history.empty()) {
+    uint64_t ad_shown_timestamp = ads_shown_history.front().timestamp_in_seconds;
+    uint64_t boot_timestamp = Time::NowInSeconds() -
+        static_cast<uint64_t>(base::SysInfo::Uptime().InSeconds());
+    if (ad_shown_timestamp <= boot_timestamp) {
+      notifications_->RemoveAll(false);
+    }
+  }
+}
+
+void AdsImpl::RemoveAllNotificationsAfterUpdate() {
+  std::string current_version_code (base::android::BuildInfo::GetInstance()->package_version_code());
```

Comment:
Versions should be retrieved from the client and not within the library. Please see GetClientInfo and add a version so this can be used cross platform

Reply:
NONCRITICAL
Covered by: brave/brave-browser#6235 (8)